blueocean-personalization/src/main/js/components/FavoritePipeline.jsx

```diff
@@ -79,13 +79,13 @@ export class FavoritePipeline extends Component {
             return null;
         }
 
-        return (
+        return (!(this.props.pipeline.branchNames && !this.props.pipeline.branchNames.length) ? (
```

Comment:
Alternatively you should be able to use `cond1 && cond2 && <ReactComponent>` instead of the ternary form, might read slightly better.